diff -ruNp orig/screen.c new/screen.c
--- orig/screen.c	2016-09-14 10:14:48.300290829 +0200
+++ new/screen.c	2016-09-14 10:16:19.119329639 +0200
@@ -694,7 +694,7 @@ char **av;
 		    strncpy(screenterm, *av, MAXTERMLEN);
 		    screenterm[MAXTERMLEN] = '\0';
 		  } else
-		    Panic(0, "-T: terminal name too long. (max. 20 char)");
+		    Panic(0, "-T: terminal name too long. (max. %d char)", MAXTERMLEN);
 		  nwin_options.term = screenterm;
 		  break;
 		case 'q':

