--- screen-4.0.3/extern.h~	2013-08-11 01:31:36.253720170 +0200
+++ screen-4.0.3/extern.h	2013-08-11 01:31:37.750722783 +0200
@@ -146,7 +146,9 @@ extern void  CloseDevice __P((struct win
 #ifdef ZMODEM
 extern void  zmodem_abort __P((struct win *, struct display *));
 #endif
+#ifndef __UCLIBC__
 extern void  execvpe __P((char *, char **, char **));
+#endif
 
 /* utmp.c */
 #ifdef UTMPOK
--- screen-4.0.3/sched.h~	2013-08-11 01:31:18.594689343 +0200
+++ screen-4.0.3/sched.h	2013-08-11 01:31:24.719700035 +0200
@@ -1,3 +1,5 @@
+#ifndef __SCHED_H
+#define __SCHED_H
 /* Copyright (c) 1993-2002
  *      Juergen Weigert (jnweiger@immd4.informatik.uni-erlangen.de)
  *      Michael Schroeder (mlschroe@immd4.informatik.uni-erlangen.de)
@@ -22,6 +24,8 @@
  * $Id: sched.h,v 1.1.1.1 1993/06/16 23:51:13 jnweiger Exp $ FAU
  */
 
+#include <sys/time.h>
+
 struct event
 {
   struct event *next;
@@ -41,3 +45,4 @@ struct event
 #define EV_READ		1
 #define EV_WRITE	2
 #define EV_ALWAYS	3
+#endif
--- screen-4.0.3/window.c~	2013-08-11 01:31:40.546727664 +0200
+++ screen-4.0.3/window.c	2013-08-11 01:31:42.087730354 +0200
@@ -1391,6 +1391,7 @@ char **args, *ttyn;
   return pid;
 }
 
+#ifndef __UCLIBC__
 void
 execvpe(prog, args, env)
 char *prog, **args, **env;
@@ -1436,6 +1437,7 @@ char *prog, **args, **env;
   if (eaccess)
     errno = EACCES;
 }
+#endif
 
 #ifdef PSEUDOS
 
