diff -uraN screen-5.0.2/termcap.c omv-screen-5.0.2/termcap.c
--- screen-5.0.2/termcap.c	2026-07-11 14:23:46.000000000 +0200
+++ omv-screen-5.0.2/termcap.c	2026-07-21 22:30:43.680556471 +0200
@@ -63,7 +63,7 @@
 static int Termcaplen;
 static int tcLineLen;
 char Term[MAXSTR + 5];		/* +5: "TERM=" */
-char screenterm[MAXTERMLEN + 1];	/* new $TERM, usually "screen" */
+char screenterm[MAXTERMLEN + 1];	/* new $TERM, usually "screen-256color" */
 
 char *extra_incap, *extra_outcap;
 
@@ -674,12 +674,12 @@
 	}
 	if ((s = getenv("SCREENCAP")) && strlen(s) < TERMCAP_BUFSIZE) {
 		sprintf(Termcap, "TERMCAP=%s", s);
-		strcpy(Term, "TERM=screen");
+		strcpy(Term, "TERM=screen-256color");
 		return Termcap;
 	}
 	Termcaplen = 0;
 	if (*screenterm == '\0' || strlen(screenterm) > MAXSTR - 3) {
-		strncpy(screenterm, "screen", MAXTERMLEN);
+		strncpy(screenterm, "screen-256color", MAXTERMLEN);
 		screenterm[MAXTERMLEN] = '\0';
 	}
 	do {
