diff -up screen-4.9.0/braille_tsi.c.omv~ screen-4.9.0/braille_tsi.c
--- screen-4.9.0/braille_tsi.c.omv~	2022-02-02 05:56:30.637761031 +0100
+++ screen-4.9.0/braille_tsi.c	2022-02-02 05:56:36.178783105 +0100
@@ -49,7 +49,7 @@ static int  tsi_line_type; /* indicates
 
 static int  display_status_tsi __P((void));
 static int  write_line_tsi __P((char*, int, int));
-static void buttonpress_tsi __P((structkey2rc*));
+//static void buttonpress_tsi __P((structkey2rc*));
 static void buttonpress_navigator_40 __P((void));
 static void buttonpress_powerbraille_40 __P((void));
 static void buttonpress_powerbraille_80 __P((void));
