--- screen-4.6.0/doc/screen.texinfo.0007~	2017-06-30 22:48:50.032386929 +0200
+++ screen-4.6.0/doc/screen.texinfo	2017-06-30 22:49:19.489741666 +0200
@@ -5796,8 +5796,8 @@ Contributors @*
 ============
 
 @verbatim
-     Thomas Renninger <treen@suse.com>,
-     Axel Beckert <abe@deuxchevaux.org>,
+     Thomas Renninger <treen@@suse.com>,
+     Axel Beckert <abe@@deuxchevaux.org>,
      Ken Beal <kbeal@@amber.ssd.csd.harris.com>,
      Rudolf Koenig <rfkoenig@@informatik.uni-erlangen.de>,
      Toerless Eckert <eckert@@informatik.uni-erlangen.de>,
